lib/smart_proxy_ansible/plugin.rb

```diff
@@ -6,11 +8,35 @@ class Plugin < Proxy::Plugin
       settings_file 'ansible.yml'
       plugin :ansible, Proxy::Ansible::VERSION
       default_settings :ansible_dir => Dir.home,
-                       :ansible_environment_file => '/etc/foreman-proxy/ansible.env'
+                       :ansible_environment_file => '/etc/foreman-proxy/ansible.env',
+                       :vcs_integration => true
                        # :working_dir => nil
 
+      load_programmable_settings do |settings|
+        mutable_roles_path = '/var/lib/foreman-proxy/ansible/roles'
+        system_roles_path = '/etc/ansible/roles:/usr/share/ansible/roles'
```

Comment:
These should be settings because in development environments or containers you may want to use different paths.

I've been thinking about some constants in the `Proxy` module, but the other day I realized that the `Proxy::Plugin` API is an even better place. That can combine those constants with the specific plugin name. So you could get a `state_dir(path)` method that would evaluate `state_dir('roles')` to `/var/lib/foreman-proxy/ansible/roles` in production. I thought I had some PR for that, but can't find the code I wrote. Essentially, I'd base it on the table in https://www.freedesktop.org/software/systemd/man/latest/systemd.exec.html#RuntimeDirectory=

I'd very much welcome PRs that implement this. It's a recurring problem in plugins.

Reply:
I have now added these fields as arguments that may be passed in the configuration file. 
The mutable_roles_path is constructed `StateDirectory + :plugin_name + roles` 
I also added `runtime, state, cache, logs and config_dir` methods. Those just construct the respective paths from an environment-variable and the plugin name. The env-variables conform to the systemd table you linked, so they can be passed easily.
If this is what you had in mind, I could create a merge-request to merge the `xxxx_dir`-methods into [smart-proxy/lib/proxy/plugin.rb](https://github.com/theforeman/smart-proxy/blob/develop/lib/proxy/plugin.rb) for reusability.